File: backend/app/schemas/wishlist.py
"""Wishlist Pydantic schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.schemas.product import ProductResponse


class WishlistItemResponse(BaseModel):
    """Wishlist item with full product details."""

    id: int
    product_id: int
    product: ProductResponse
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class WishlistToggleResponse(BaseModel):
    """Response after toggling a wishlist item."""

    is_wishlisted: bool

File: frontend/src/types/wishlist.ts
/**
 * Wishlist TypeScript interfaces
 */

import type { Product } from "./product";

export interface WishlistItem {
  id: number;
  product_id: number;
  product: Product;
  created_at: string;
}

export interface WishlistToggleResponse {
  is_wishlisted: boolean;
}
